src/ir/auth_logic/lowering_ast_datalog.cc

```diff
@@ -197,8 +193,7 @@ LoweringToDatalogPass::FactToDLIR(const Principal& speaker, const Fact& fact) {
                 "says_", speaker,
                 PushPrincipal("canSay_", fresh_principal, inner_fact_dlir));
             auto rhs = {fresh_prin_says_inner, speaker_says_fresh_cansay_inner};
-            datalog::DLIRAssertion generated_rule(
-                datalog::DLIRCondAssertion(lhs, rhs));
+            datalog::Rule generated_rule(lhs, rhs);
             gen_rules.push_back(generated_rule);
```

Comment:
Get rid of `generated_rule` and inline like so:

```C++
gen_rules.push_back(datalog::Rule(lhs, rhs));
```

I'd take a look at all of these locals which are used only once and see how many you can eliminate by inlining without sacrificing readability. I suspect many of them should be `std::move`d even if the temporaries should stay.